src/pages/HomePage/index.tsx

```diff
@@ -1,20 +1,92 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
 import React from "react";
+import { flushSync } from "react-dom";
 import FiltersTab from "../../components/FiltersTab";
 import Footer from "../../components/Footer";
 import Navbar from "../../components/Navbar";
 import MoviesList from "../../components/MoviesList";
 import SearchBar from "../../components/SearchBar";
-import style from "./styles.module.scss";
 import { movies } from "../../mockData";
 import ErrorBoundary from "../../components/ErrorBoundary";
 import { Box } from "@mui/system";
 import Button from "@mui/material/Button";
+import { Modal } from "@mui/material";
+import AddMovieModal from "../../components/AddMovieModal";
+import { Movie } from "../../constants";
+import style from "./styles.module.scss";
+
+class Home extends React.Component<
+  any,
+  { isOpen: boolean; activeTab: string; movies: Movie[]; sortOption: string }
+> {
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      isOpen: false,
+      activeTab: "",
+      movies: movies,
+      sortOption: "",
+    };
+  }
+
+  componentDidMount(): void {
+    this.sortByGenre();
+  }
+
+  handleOpen = () => this.setState({ isOpen: true });
+
+  handleClose = () => this.setState({ isOpen: false });
+
+  setActiveTab = (activeTab: string) => {
+    flushSync(() => {
+      this.setState({ activeTab });
+    });
+
+    this.sortByGenre();
+  };
+
+  setSortByOption = (option: string) => {
+    flushSync(() => {
+      this.setState({ sortOption: option });
+    });
+
+    this.sortByGenre();
+  };
+
+  sortByGenre = () => {
+    const sortedMovies = movies.filter((movie) =>
+      movie.genre.toLowerCase().includes(this.state.activeTab.toLowerCase())
+    );
+    if (this.state.sortOption === "release date") this.sortByDate(sortedMovies);
+    else if (this.state.sortOption === "duration")
+      this.sortByDuration(sortedMovies);
+    else this.sortByRating(sortedMovies);
+  };
+
+  sortByRating = (movies: Movie[]) => {
+    this.setState({
+      movies: movies.sort((a, b) => +b.imdbRating - Number(a.imdbRating)),
```

Comment:
we probably need a comment here with explanation, also why both "+" and Number() used ?